shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/domain/usecase/GetFeedingPointByPriorityUseCase.kt

```diff
@@ -0,0 +1,66 @@
+package com.epmedu.animeal.feeding.domain.usecase
+
+import com.epmedu.animeal.feeding.domain.model.FeedingPoint
+import com.epmedu.animeal.feeding.domain.model.enum.AnimalState
+import com.epmedu.animeal.feeding.domain.repository.FeedingPointRepository
+import com.epmedu.animeal.feeding.presentation.model.MapLocation
+import com.epmedu.animeal.feeding.presentation.model.MapLocation.Companion.toPoint
+import com.epmedu.animeal.foundation.tabs.model.AnimalType
+import com.epmedu.animeal.geolocation.location.distanceInKmTo
+import com.epmedu.animeal.geolocation.location.isNearTo
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class GetFeedingPointByPriorityUseCase(
+    val feedingPointRepository: FeedingPointRepository
+) {
+
+    /*
+     * Filters list of FeedingPoint by priority defined at EPMEDU-594:
+     * - 1st: Status = Red, is favorite and is near to user location
+     * - 2nd: Status = Red, is favorite
+     * - 3rd: Status = Red, is not favorite but is near to user location
+     */
+    operator fun invoke(
+        type: AnimalType,
+        userLocation: MapLocation,
+    ): Flow<FeedingPoint?> {
```

Comment:
Let's return not nullable `FeedingPoint` to avoid using let by adding `filterNotNull` in the chain below